### PR TITLE
Add ilastik content types

### PR DESCRIPTION
**Pull request by MaaikevS**
@lzehl @UlrikeS91 

I added the content types for Ilastik based on Issue #296 
If you find that anything is missing or incorrect, let me know.

### CODE REVIEW

**Comment by UlrikeS91** [comment]
> @lzehl @UlrikeS91
> 
> I added the content types for Ilastik based on Issue #296 If you find that anything is missing or incorrect, let me know.

I'm not an expert. So, I won't involve myself. I added @jagru20, @apdavison and @lzehl as reviewers. I don't believe that all are required, but this way whoever has time first, can review the PR.

File: instances/data/contentTypes/ilastik+csv.jsonld [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "@context": {
+    "@vocab": "https://openminds.ebrains.eu/vocab/"
+  },
+  "@id": "https://openminds.ebrains.eu/instances/contentTypes/application/vnd.ilastik+csv",
```
Comment (jagru20) [inline instances/data/contentTypes/ilastik+csv.jsonld:5]: according to the naming convention proposed by @apdavison the +suffix should be used for container formats. As far as I understand container formats, they are formats, that contain several other formats (like .zip files) therefore I assume +csv to be not correct and would just cut it off. But I am not 100% sure, @apdavison could you please give your thought on this?

Comment (jagru20) [reply instances/data/contentTypes/ilastik+csv.jsonld:5]: I think this would then also apply to others in this PR, as +tiff, +png etc.

Comment (jagru20) [reply instances/data/contentTypes/ilastik+csv.jsonld:5]: After one more thought about it, I would suggest naming it as subtypes like `application/vnd.ilastik.csv`

File: instances/data/contentTypes/ilastik+hdf5.jsonld [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+	"@context" : {
+		"@vocab": "https://openminds.ebrains.eu/vocab/"
+	},
+	"@id": "https://openminds.ebrains.eu/instances/contentTypes/application/vnd.ilastik+hdf5",
```
Comment (jagru20) [inline instances/data/contentTypes/ilastik+hdf5.jsonld:5]: based on [this comment ](https://github.com/HumanBrainProject/openMINDS_core/pull/320#issuecomment-1073669832) it should be `/vnd.ilastik.purpose1+hdf5` if we can distinguish the purpose the files with .ilp extensions are for.

File: instances/data/contentTypes/ilastik+csv.jsonld (hunk 1 above)
Comment (jagru20) [inline instances/data/contentTypes/ilastik+csv.jsonld]: based on [this comment ](https://github.com/HumanBrainProject/openMINDS_core/pull/320#issuecomment-1073669832) it should be `/vnd.ilastik.purpose1+csv` if we can distinguish the purpose the ilastik files with .csv extensions are for.

**Review by jagru20 — CHANGES_REQUESTED** [review]
If we can identify the purpose of the .csv and .ilp files, we should append them as the last subtype. I think the rest is ok.

**Comment by lzehl** [comment]
@k-dominik could you help us figuring out the open issues in the PR? For your information: openMINDS content types are inspired by IANA.org media types to define software (version) specific file types that can go beyond generic formats (e.g. a software specific json). We are trying to register all input and output content types of Ilastik in this PR. (feel free to pin other persons of the Ilastik development team if their input is required). Thank you in advance :)

**Review by k-dominik — COMMENTED** [review]
Hello all involved (@MaaikevS, @UlrikeS91, @jagru20, and @lzehl).

first of all I think it is a really cool effort to systematically attach some meaning to certain outputs of software.

as for the two different _purposes_ for csv/hdf5 one could probably use a term like _feature table_ or _result table_ or something like it.

Maybe as background: ilastik encapsulates different workflows, e.g. Pixel Classification that produces semantic segmentation (with output being either "simple segmentation" or "probabilities"), or Object Classification that does instance segmentation, produces output images like "Object Predictions", "Object Probabilites", "Object Identities", as well as the tables in csv/hdf5 format.

Do you maybe think it makes sense to add the workflow to the schema? E.g. `...image/vnd.ilastik.<workflowname>.probabilties+hdf5` ?

nitbit: usually ilastik is never capitalized, but it seems to be in the synonym field.

**Comment by lzehl** [comment]
@k-dominik thank you so much for the clarifications.  I have a question in order to answer your question:
Are the internal file structure of the hdf5 files workflow specific (meaning do I have to load/read them differently)? 

If yes, I would indeed split the content types up and follow the naming convention you suggested (`...image/vnd.ilastik.<workflowname>.probabilties+hdf5`). If no, I would keep the workflow out of the content type definition (`...image/vnd.ilastik.probabilties+hdf5`). 

I also have another favor to ask: could you give us a short (1-2 sentence) description we could add for each ilastik content type?
- `...image/vnd.ilastik.probabilties+hdf5`
- `...image/vnd.ilastik.probabilties+csv`
- `...image/vnd.ilastik.prediction+hdf5`
- `...image/vnd.ilastik.prediction+jpeg`
- `...image/vnd.ilastik.prediction+png`
- `...image/vnd.ilastik.prediction+tiff-stack`
- `...image/vnd.ilastik.prediction+tiff`
- `...image/vnd.ilastik.segmentations+hdf5`
- `...image/vnd.ilastik.segmentations+jpeg`
- `...image/vnd.ilastik.segmentations+png`
- `...image/vnd.ilastik.segmentations+tiff-stack`
- `...image/vnd.ilastik.segmentations+tiff`

**Comment by k-dominik** [comment]
Okay I see - I don't think adding the workflow to the predictions type is necessary - while there could be a _semantic_ difference depending on the workflow, it will in any case be an nd-image.

For the 1-2 sentence description - can I assume that describing _probabilities_/_predictions_ (terms we use interchangeably) and _segmentation_ would be enough?

**Comment by lzehl** [comment]
@k-dominik you mean we can only use "prediction" or only use "probabilties" in the content types above? I though the ...image/vnd.ilastik.probabilties+hdf5 and ...image/vnd.ilastik.prediction+hdf5 is not the same, or is it? Maybe we can have a short call to clarify this issue ?

**Comment by lzehl** [comment]
RESULT discussion with @k-dominik :
...image/vnd.ilastik.hdf5 OR
...image/vnd.ilastik.image+hdf5
+ Reduce to one, because the internal structure does not change.

...application/vnd.ilastik.hdf5 OR
...application/vnd.ilastik.features+hdf5
+ contains feature table / object classification (can contain more than the csv)

...application/vnd.ilastik.csv OR
...application/vnd.ilastik.features+csv
+ contains feature table

**Comment by lzehl** [comment]
For the workflow project files  (with extension: .ilp):

...application/vnd.ilastik.project.WORKFLOW+hdf5 OR
...application/vnd.ilastik.WORKFLOW+hdf5

**Comment by k-dominik** [comment]
we could start with the workflows

* `Pixel Classification`, and
* `Object Classification`.

**Comment by lzehl** [comment]
SUMMARY with feedback from @apdavison & @k-dominik  : 

+ `image/vnd.ilastik.hdf5` (input/output)
+ `application/vnd.ilastik.features+hdf5` (output)
+ `application/vnd.ilastik.features+csv` (output)
+ `application/vnd.ilastik.pixelclassification+hdf5` (input/output)
+ `application/vnd.ilastik.objectclassification+hdf5` (input/output)

Note: additional input / output for ilastik are the general formats for png, jpeg, tiff, and tiff-stack (because they are not specially formatted/modified by ilastik in any way).

**Comment by lzehl** [comment]
@MaaikevS sorry I forgot to ask: do you want to do the changes for this PR or should I take over?

**Comment by lzehl** [comment]
@MaaikevS I think I'm not able to take make the needed edits on your branch. Could you do the final changes or should I close this PR and make a new one?

Here a summary of what needs to be done:

Please reduce the PR to the following content types
    image/vnd.ilastik.hdf5 (input/output)
    application/vnd.ilastik.features+hdf5 (output)
    application/vnd.ilastik.features+csv (output)
    application/vnd.ilastik.pixelclassification+hdf5 (input/output)
    application/vnd.ilastik.objectclassification+hdf5 (input/output)

Please adapt the atIDs of the content types to this (new) convention:
`"@id": "https://openminds.ebrains.eu/instances/contentTypes/image_vnd.ilastik.hdf5"`
`"@id": "https://openminds.ebrains.eu/instances/contentTypes/application_vnd.ilastik.features+hdf5"`
`"@id": "https://openminds.ebrains.eu/instances/contentTypes/application_vnd.ilastik.features+csv"`
`"@id": "https://openminds.ebrains.eu/instances/contentTypes/application_vnd.ilastik.pixelclassification+hdf5"`
`"@id": "https://openminds.ebrains.eu/instances/contentTypes/application_vnd.ilastik.objectclassification+hdf5"`

Please also adapt the file names to this (new) convention:
    image_vnd.ilastik.hdf5.jsonld
    application_vnd.ilastik.features+hdf5.jsonld
    application_vnd.ilastik.features+csv.jsonld
    application_vnd.ilastik.pixelclassification+hdf5.jsonld
    application_vnd.ilastik.objectclassification+hdf5.jsonld

**Comment by lzehl** [comment]
@MaaikevS I'm closing your PR without merging due to the last discussions on this topic and the work connected to this. Thanks a lot for pushing this in the past.

**Pull request closed without merging**